src/plugins/intellisense/latex-parsing.tsx

```diff
@@ -0,0 +1,211 @@
+import { MathQuillField } from "components";
+import { ExpressionAug } from "plugins/text-mode/aug/AugState";
+
+export function mapAugAST(
+  node: ExpressionAug["latex"],
+  callback: (node: ExpressionAug["latex"]) => void
+) {
+  function map(x: any) {
+    if (Array.isArray(x)) {
+      for (const child of x) {
+        map(child);
+      }
+    }
+
+    if (typeof x === "object") {
+      if (typeof x.type === "string") callback(x);
+
+      for (const [_, v] of Object.entries(x)) {
+        map(v);
+      }
+    }
+  }
+
+  map(node);
+}
+
+export function getController(mq: MathQuillField) {
+  return mq.__controller as MQController;
+}
+
+export function mqKeystroke(mq: MathQuillField, keystroke: string) {
+  mq.keystroke(keystroke);
+}
+
+export interface MQController {
+  cursor: MQCursor;
+}
+
+export interface MQCursor {
+  parent?: MQCursor;
+  latex?: () => string;
+  [-1]: MQCursor | undefined;
+  [1]: MQCursor | undefined;
+  cursorElement?: HTMLElement;
+  ctrlSeq?: string;
+  _el?: HTMLElement;
+}
+
+export const identRegex = /([a-zA-Z]|\\[a-zA-Z]+) *(_\{[a-zA-Z0-9 ]*\})?/g;
```

Comment:
Is `\operatorname` ever relevant where this is used?

Reply:
`identRegex` as it's currently implemented is only designed to detect user-generated identifiers, which as far as I know don't use `\operatorname` (correct me if I'm wrong). I'm considering modifying it (and by extension modifying the rest of the codebase) to eventually support builtin identifiers as well.

Reply:
You can write `\\operatorname{min}_{xy}`, though I don't see it much (and maybe Desmos will eventually disallow it /shrug).

Reply:
that is absolutely terrifying

Reply:
I'm gonna pre-emptively resolve this since I think we decided it's out of scope.